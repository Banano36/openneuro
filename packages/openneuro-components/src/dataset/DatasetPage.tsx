import React from 'react'

export interface DatasetPageProps {
  renderSidebar: () => React.ReactNode
  renderValidationBlock: () => React.ReactNode
  renderHeader: () => React.ReactNode
  renderAlert: () => React.ReactNode
  renderHeaderMeta: () => React.ReactNode
  renderToolButtons: () => React.ReactNode
  renderReadMe: () => React.ReactNode
}

export const DatasetPage = ({
  renderSidebar,
  renderValidationBlock,
  renderHeader,
  renderAlert,
  renderHeaderMeta,
  renderToolButtons,
  renderReadMe,
}: DatasetPageProps) => {
  return (
    <>
      {renderHeader()}
      {renderAlert()}
      <div className="container">
        <div className="grid grid-between">
          <div className="col">{renderHeaderMeta()}</div>
        </div>
      </div>
      <div className="container">
        <div className="grid grid-between">
          <div className="col col-8">
            {renderValidationBlock()}
            {renderToolButtons()}
            {renderReadMe()}
          </div>
          <div className="col sidebar"> {renderSidebar()}</div>
        </div>
      </div>
    </>
  )
}
