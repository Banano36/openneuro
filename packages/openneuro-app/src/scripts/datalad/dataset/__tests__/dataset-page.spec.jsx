import React from 'react'
import { shallow } from 'enzyme'
import DatasetPage from '../dataset-page.jsx'

describe('DatasetPage component', () => {
  it('renders with common props', () => {
    expect(
      shallow(
        <DatasetPage
          dataset={{
            id: '1234',
            snapshots: [],
            draft: { modified: '2018-10-10' },
          }}
        />,
      ),
    ).toMatchSnapshot()
  })
  it('hides sidebar when clicked', () => {
    const wrapper = shallow(
      <DatasetPage
        dataset={{
          id: '1234',
          snapshots: [],
          draft: { modified: '2018-10-10' },
        }}
      />,
    )
    expect(wrapper.find('.dataset-container').hasClass('open')).toBe(true)
    wrapper.find('.show-nav-btn').simulate('click')
    expect(wrapper.find('.dataset-container').hasClass('open')).toBe(false)
  })
})
