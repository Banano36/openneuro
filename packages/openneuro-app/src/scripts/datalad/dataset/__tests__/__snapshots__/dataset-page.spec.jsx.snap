// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetPage component renders with common props 1`] = `
<div
  className="page dataset"
>
  <div
    className="open dataset-container"
  >
    <withRouter(LeftSidebar)
      datasetId="1234"
      draftModified="2018-10-10"
      snapshots={Array []}
    />
    <span
      className="show-nav-btn"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-angle-double-left"
      />
    </span>
    <withRouter(DatasetTools)
      dataset={
        Object {
          "draft": Object {
            "modified": "2018-10-10",
          },
          "id": "1234",
          "snapshots": Array [],
        }
      }
    />
    <div
      className="fade-in inner-route dataset-route light"
    >
      <div
        className="clearfix dataset-wrap"
      >
        <div
          className="dataset-animation"
        >
          <div
            className="col-xs-12 dataset-inner"
          >
            <DatasetMain
              dataset={
                Object {
                  "draft": Object {
                    "modified": "2018-10-10",
                  },
                  "id": "1234",
                  "snapshots": Array [],
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
